core/unit_test/cuda/TestCuda_InterOp_StreamsMultiGPU.cpp

```diff
@@ -129,10 +131,14 @@ TEST_F(TEST_CATEGORY_FIXTURE(multi_gpu), stream_sync_semantics_raw_cuda) {
     // Allocate data.
     int *value;
     int *check;
+    void *value_void_ptr = nullptr;
+    void *check_void_ptr = nullptr;
     KOKKOS_IMPL_CUDA_SAFE_CALL(
-        cudaMallocHost(reinterpret_cast<void **>(&value), 1 * sizeof(int)));
+        cudaMallocHost(&value_void_ptr, 1 * sizeof(int)));
     KOKKOS_IMPL_CUDA_SAFE_CALL(
-        cudaMallocHost(reinterpret_cast<void **>(&check), 1 * sizeof(int)));
+        cudaMallocHost(&check_void_ptr, 1 * sizeof(int)));
+    value = static_cast<int *>(value_void_ptr);
+    check = static_cast<int *>(check_void_ptr);
```

Comment:
I am not loving the interleaving of operations as I see the three statements to avoid the reinterpret_cast as an indissociable operation.  Not blocking though.